tests/shared/src/test/scala/scala/meta/tests/parsers/TermSuite.scala

```diff
@@ -112,6 +112,18 @@ class TermSuite extends ParseSuite {
     assertTerm("f(x = xs: _*)") {
       Term.Apply(Term.Name("f"), List(Assign(Term.Name("x"), Repeated(Term.Name("xs")))))
     }
+    val Term.Apply.Until_4_6_0(
+      Term.Name("f"),
+      List(Assign(Term.Name("x"), Repeated(Term.Name("xs"))))
+    ) = term("f(x = xs: _*)")
+    val Term.Apply.After_4_6_0(
```

Comment:
Considering the constraint "[Every Scala program linkable against x.y.z1 must be linkable against x.y.z2.](https://github.com/scalameta/scalameta/blob/main/COMPATIBILITY.md)", there shouldn't be any change in the 4.6.x releases, so we could omit the patch version
```suggestion
    val Term.Apply.After_4_6(
```

Reply:
Is `Since` less ambiguous?
```suggestion
    val Term.Apply.Since_4_6(
```

Reply:
@bjaglin while it might be correct in terms of semver, i don't want to overload this logic with those assumptions/requirements. i am simply using the version, as-is, without any semantic parsing, that is specified in the `@newField` or `@replacedField` annotation on a tree field.

Reply:
sorry, didn't comment on `Since`. to me, "since" implies `>=` whereas this is strictly `>` which is what I tried to convey with "after".

Reply:
Oh, I didn't realize `replacedField` had a `until` parameter, unlike `newField` and `replacesFields` which have a `since` one. Is there a reason for that?

I find `Since_4_7_0` more useful than `After_4_6_0` (which only tells me this syntax won't build against 4.6.0). Also, if I understand correctly, having the `since` would allow for a more precise/canonical deprecation message (see other comment above).

Reply:
1. this reminds me, I should rename `since` parameter in `newField` and `replacesFields` to `after`.
2. `replacedField` means this field was replaced after the given version and is no longer available, the parameter shows the last version it was present in
3. the other two are marked with the version after which they were added.

when we add the code which changes these fields, the only thing we know for a fact is which version was the last released one (and is exactly how `-SNAPSHOT` version is computed). similarly, we have a check that verifies that none of the version tags are later than the current build version. of course, this check will not be possible otherwise.

i can make an educated *guess* about the next release tag but since it hasn't happened yet, it's just a guess. and here i prefer not to do that :) (sort of like some parks in this area which state that they close 30 minutes _before_ dusk.)

if you change your code to depend on version 4.7.4, does it matter if the unapply is called `After_4_6_0` or `Since_4_7_0`? neither of them is `AsOf_4_7_4` anyway...

Reply:
> i can make an educated guess about the next release tag but since it hasn't happened yet, it's just a guess.

Considering https://github.com/scalameta/scalameta/blob/main/COMPATIBILITY.md, any new or replaced field needs to be introduced in the next minor release (or major one, but I believe these are so rare that they could be considered the exception), so that helps educating your guess further ;) On a side note, forward compatibility within patch versions does not seem to be checked with MiMa though (but I believe it would be possible to run it in forward mode?).

> if you change your code to depend on version 4.7.4, does it matter if the unapply is called After_4_6_0 or Since_4_7_0? neither of them is AsOf_4_7_4 anyway...

Agreed! It's more that from a consumer perspective, it just takes a bit of thinking to understand why you get a deprecation warning mentioning 4.6.0 when bumping from 4.6.0 to 4.7.x and end up having code mentioning the *prior* version to get rid of it.

I wish there was more tooling available to help developers provide a "since-centric" view (since that's what deprecation annotation uses). From the projects I know, it's mostly handled manually (based on educated guesses).